transport/src/main/java/io/netty/channel/socket/InternetProtocolFamily.java

```diff
@@ -38,4 +44,52 @@
     public Class<? extends InetAddress> addressType() {
         return addressType;
     }
+
+    /**
+     * Returns the <a href="http://www.iana.org/assignments/address-family-numbers/address-family-numbers.xhtml">
+     *     address number</a> of the family.
+     */
+    public int addressNumber() {
+        return addressNumber;
+    }
+
+    /**
+     * Returns the {@link InetAddress} that represent the {@code LOCALHOST} for the family.
+     */
+    public InetAddress localHost() {
+        return localHost;
+    }
+
+    private static InetAddress localHost(Class<? extends InetAddress> addressType) {
+        if (addressType.isAssignableFrom(Inet4Address.class)) {
+            return NetUtil.LOCALHOST4;
+        }
+        if (addressType.isAssignableFrom(Inet6Address.class)) {
+            return NetUtil.LOCALHOST6;
+        }
+        throw new Error();
+    }
+
+    private static int addressNumber(Class<? extends InetAddress> addressType) {
+        if (addressType.isAssignableFrom(Inet4Address.class)) {
+            return 1;
+        }
+        if (addressType.isAssignableFrom(Inet6Address.class)) {
+            return 2;
+        }
+        throw new IllegalArgumentException("addressType " + addressType + " not supported");
+    }
+
+    /**
+     * Returns the {@link InternetProtocolFamily} for the given {@link InetAddress}.
+     */
+    public static InternetProtocolFamily familyOf(InetAddress address) {
```

Comment:
I kind of dislike using just something like `of(...)` as it may easily conflict with other methods when the user uses static imports. WDYT ?